WoWsShipBuilder.Web/Dialogs/BuildStringInputDialog.razor

```diff
@@ -0,0 +1,108 @@
+﻿@using WoWsShipBuilder.Core.Builds
+@using System.Net
+@inject ILocalizer Localizer
+
+<MudDialog Style="width: 600px">
+    <DialogContent>
+        <MudFocusTrap DefaultFocus="DefaultFocus.FirstChild">
+            <MudTextField @bind-Value="buildString" Label="@Localizer.GetAppLocalization(nameof(Translation.BuildStringInputDialog_EnterBuildString)).Localization" Immediate Validation="@(new Func<string, string?>(ValidateBuildString))"/>
+        </MudFocusTrap>
+    </DialogContent>
+    <DialogActions>
+        <MudButton OnClick="@Cancel">
+            @Localizer.GetAppLocalization(nameof(Translation.Dialog_Cancel)).Localization
+        </MudButton>
+        <MudButton OnClick="@OnConfirm" Disabled="!validBuild">
+            @Localizer.SimpleAppLocalization(nameof(Translation.ShipSelectionWindow_ConfirmButton))
+        </MudButton>
+    </DialogActions>
+</MudDialog>
+
+@code {
+
+    [CascadingParameter]
+    MudDialogInstance MudDialog { get; set; } = new();
+
+    [Parameter]
+    public string SelectedShipIndex { get; set; } = default!;
+    
+    private string buildString = string.Empty;
+
+    private bool validBuild;
+
+    private void Cancel()
+    {
+        MudDialog.Cancel();
+    }
+
+    private void OnConfirm()
+    {
+        string buildStr;
+        if (buildString.Contains("build="))
```

Comment:
The variable name "buildString" implies that it only contains the unescaped build string